rules/linux/execution_abnormal_process_id_file_created.toml

```
@@ -27,11 +27,55 @@ name = "Abnormal Process ID or Lock File Created"
 note = """## Triage and analysis
 
 ### Investigating Abnormal Process ID or Lock File Created
-Detection alerts from this rule indicate that an unusual PID file was created and could potentially have alternate purposes during an intrusion.  Here are some possible avenues of investigation:
-- Run the following in Osquery to quickly identify unsual PID file size: "SELECT f.size, f.uid, f.type, f.path from file f WHERE path like '/var/run/%pid';"
-- Examine the history of this file creation and from which process it was created by using the "lsof" command.
-- Examine the contents of the PID file itself, simply by running the "cat" command to determine if the expected process ID integer exists and if not, the PID file is not legitimate.
-- Examine the reputation of the SHA256 hash from the PID file in a database like VirusTotal to identify additional pivots and artifacts for investigation."""
+
+Linux applications may need to save their process identification number (PID) for various purposes: from signaling that
+a program is running to serving as a signal that a previous instance of an application didn't exit successfully; for
+this, they use PID files. PID files contain its creator process PID in an integer value.
+
+Linux lock files are used to coordinate operations in files so that conflicts and race conditions are prevented.
+
+This rule identifies the creation of PID, lock, or reboot files in the /var/run/ directory. Attackers can masquerade
+malware, payloads, staged data for exfiltration, and more, as legitimate PID files.
+
+#### Possible investigation steps
+
+- Retrieve the file and determine if it is malicious:
+    - Check the contents of the PID files, it should only contain an integer string.
+    - Check the file type of the lock and PID files to determine if they are an executable — this is only observed in
+    malicious files.
+    - Check the size of the subject file. Legitimate PID files should be under 10 bytes.
+    - Check if the lock or PID file has high entropy. This typically indicates an encrypted payload.
+    - Examine the reputation of the SHA256 hash in the PID file. Use a database like VirusTotal to identify additional
+    pivots and artifacts for investigation.
+- Trace the file's creation to ensure it came from a legitimate or authorized process.
+- Investigate other alerts associated with the user/host during the past 48 hours.
+- Investigate any abnormal account behavior, such as command executions, file creations or modifications, and network
+connections.
+- Investigate any abnormal behavior by the subject process such as network connections, file modifications, and any
+spawned child processes.
+
+### False positive analysis
+
+- False-Positives (FP) can appear if the PID file is legitimate and holding a process ID as intended. If the PID file is
```

We don't need to explain the abbreviation, because it's not used anywhere else in this content.
```suggestion
- False positives can appear if the PID file is legitimate and holding a process ID as intended. If the PID file is
```